src/category_theory/limits/shapes/types.lean

```diff
@@ -223,4 +227,139 @@ def equalizer_limit : limits.limit_cone (parallel_pair g h) :=
+have (π left ≫ f) w = (π right ≫ g) w, from congr_fun this w,
+have f x = g y, by rw [types_comp_apply] at this; assumption,
+⟨⟨x, y⟩, this⟩
+
+@[simp] lemma pullback_lift_val (s : limits.cone (limits.cospan f g)) (w : s.X)
+  : (pullback_lift f g s w).val = ⟨s.π.app left w, s.π.app right w⟩ :=
+begin rcases s with ⟨W, ⟨π, commutes⟩⟩, refl end
+
+@[simp] lemma pullback_lift_fst (s : limits.cone (limits.cospan f g)) (w : s.X)
+  : (↑(pullback_lift f g s w) : X × Y).fst = (pullback_lift f g s w).val.fst :=
+rfl
+
+/--
+The explicit pullback given by `pullback_cone f g` is a limit.
+This is bundled with the cone itself as `pullback_limit_cone f g`.
+-/
+def pullback_limit : is_limit (pullback_cone f g) :=
+{ lift := pullback_lift f g,
+
+  fac' := λ s j, begin
+    rcases j with hbase | hleft | hright,
+    rw ←s.w inl,
+
+    -- Concise but slow version:
+    -- all_goals { funext w, simp [-subtype.val_eq_coe], },
+    -- (This can get wedged by converting val to coe: f ↑(pullback_lift f g s w).fst = s.π.app one w )
+
+    all_goals {
+      funext w,
+      rw [pullback_cone_app, types_comp_apply],
+      try { rw [types_comp_apply, cospan_map_inl] },
+      rw pullback_proj_base <|> rw pullback_proj_fst <|> rw pullback_proj_snd,
+      dsimp only [],
+      rw pullback_lift_val,
+    },
+  end,
+
+  uniq' := begin
+    intro s,
+    rintros (lift: s.X ⟶ pullback_obj f g) h_lift,
+    funext w,
+    rw pullback_cone_app at h_lift,
+    have h_fst : (lift w).val.fst = (pullback_lift f g s w).val.fst,
+      by rw [pullback_lift_val, ←h_lift left, types_comp_apply, pullback_proj_fst],
+    have h_snd : (lift w).val.snd = (pullback_lift f g s w).val.snd,
+      by rw [pullback_lift_val, ←h_lift right, types_comp_apply, pullback_proj_snd],
+    ext, exact h_fst, exact h_snd,
+  end,
+}
+
+/--
+The explicit pullback in the category of types, bundled up as a `limit_cone`
+for given `f` and `g`.
+-/
+def pullback_limit_cone (f : X ⟶ Z) (g : Y ⟶ Z) : limits.limit_cone (cospan f g) :=
+{ cone := pullback_cone f g,
+  is_limit := pullback_limit f g, }
+
+/--
+The pullback cone given by the instance `has_pullbacks (Type u)` is isomorphic to the
+explicit pullback cone given by `pullback_limit_cone`.
+-/
+noncomputable def pullback_cone_iso_pullback : limit.cone (cospan f g) ≅ pullback_cone f g :=
+(limit.is_limit _).unique_up_to_iso (pullback_limit f g)
+
+/--
+The pullback given by the instance `has_pullbacks (Type u)` is isomorphic to the
+explicit pullback object given by `pullback_limit_obj`.
+-/
+noncomputable def pullback_iso_pullback : pullback f g ≅ pullback_obj f g :=
+cones.hom_iso_of_cone_iso (pullback_cone_iso_pullback f g)
+
+@[simp] lemma pullback_fst'
+  : limits.pullback.fst = (pullback_iso_pullback f g).hom ≫ pullback_fst f g :=
+eq.symm $ (pullback_cone_iso_pullback f g).hom.w _
+
+@[simp] lemma pullback_snd'
+  : limits.pullback.snd = (pullback_iso_pullback f g).hom ≫ pullback_snd f g :=
+eq.symm $ (pullback_cone_iso_pullback f g).hom.w _
```

Comment:
These are definitely the wrong way wrong as simp lemmas!

Reply:
Flipped them around.